e_commerce_app/LoginView/LoginViewController.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  LoginViewController.swift
+//  e_commerce_app
+//
+//  Created by Bhavin Kapadia on 2022-09-04.
+//
+
+import Foundation
+import UIKit
+import CoreData
+
+class LoginViewController: UIViewController {
+    
+    @IBOutlet weak var loginErrorLabel: UILabel?
+    @IBOutlet weak var loginUsernameField: UITextField!
+    @IBOutlet weak var loginPasswordField: UITextField!
+    
+    private let loginViewModel = LoginViewModel()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setDelegates()
+        loginErrorLabel?.alpha = 0 // hide error label on launch
+    }
+    
+    @IBAction func signInButtonTapped(_ sender: Any) {
+        validateSignInFields()
+    }
+
+    private func validateSignInFields() -> String? {
```

Comment:
This validation can be in the view model.

Reply:
And instead of returning a string, this could return an `Error`.

Reply:
addressed with recent changes in PR
 [8384a37](https://github.com/bkapadia01/ecommerce_ios_app/pull/1/commits/8384a373b9263ad4f79642138bffd9a693136d78) 
[19310db](https://github.com/bkapadia01/ecommerce_ios_app/pull/1/commits/19310dbe50d30c00129a8a781501832b3fd53b06)
https://github.com/bkapadia01/ecommerce_ios_app/pull/1/commits/d13137c26e351cc4c6716013ce77d1cbe8cd36b1